pkg/generate/app/template.go

```diff
@@ -0,0 +1,47 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/template"
+	templateapi "github.com/openshift/origin/pkg/template/api"
+)
+
+type Template struct {
+	From   string
+	Stored bool
+
+	Template *templateapi.Template
+	Env      Environment
+}
+
+type Templates []*Template
+
+func NewTemplate(from string, t *templateapi.Template, environment Environment) (*Template, error) {
+	return &Template{
+		From:     from,
+		Stored:   true,
+		Template: t,
+		Env:      environment,
+	}, nil
+}
+
+func (t *Template) Process(c client.TemplateConfigsNamespacer, namespace string) ([]runtime.Object, error) {
```

Comment:
This whole class has very little value.  It would be better represented just as a single function.